src/features/session_replay/aggregate/index.js

```diff
@@ -121,16 +120,26 @@ export class Aggregate extends AggregateBase {
         }
       }, this.featureName, this.ee)
 
-      this.waitForFlags(['sr']).then(([flagOn]) => this.initializeRecording(
+      Promise.all([
+        this.waitForFlags(['sr']),
+        this.waitForOptIn()
+      ]).then(([[flagOn]]) => this.initializeRecording(
```

Comment:
I think this is no longer necessary with the opt in being tied to the importAggregator() call now, should remove